pymongo/collection.py

```diff
@@ -2472,11 +2472,12 @@ def aggregate_raw_batches(
             raise InvalidOperation("aggregate_raw_batches does not support auto encryption")
         if comment is not None:
             kwargs["comment"] = comment
+        cursor_class = kwargs.pop("cursor_class", RawBatchCommandCursor)
```

Comment:
That won't work, because we need to control the unpacking of the raw response as well before it is consumed in `__send_message`.   Instead we could pass in a custom unpacker for both `aggregate_raw_batches()` and `find_raw_batches()` that handles setting `user_fields` and unpacking the `raw_response` using the bson stream parser.  So all cases `_unpack_response` would check for a custom unpacker and use it instead of its own logic if present.

Reply:
I want to avoid making those warty apis public in any way. Can we instead make a `aggregate_raw_batches(raw_command=True)` cursor more lazily decode the command response without inflating the documents? Perhaps a new lazier RawBSONDocument that only decodes the one field being accessed instead of the entire level of bson?

Reply:
Not without having a bson stream parser in the bson package, unless there's a way I couldn't find while attempting to do just that.

Reply:
RawBSONDocument could be a quasi stream parser. What if we changed it to lazily decode an embedded BSON array field. For example, raw_bson._inflate_bson could wrap embeded arrays in a new, and more importantly, private   _RawBSONArray class. Then RawBSONDocument would only inflate the array once a user actually requests access to that specific field.

Reply:
We are still accessing `cursor["firstBatch"]` in `__send_message`.  I think if `raw_response` was true, we'd leave "firstBatch"/"nextBack" as raw bytes, otherwise we'd inflate it as we currently do.

Reply:
Yes that's another hurdle the new cursor type needs to jump over. We can't publicly exposing the internal cursor_class as a public api. It's too unstable and will prevent us from refactoring/improving the codebase in the future. The api needs to be something as simple as:
```
for raw_command in aggregate_raw_batches(raw_command=True):
    ....
```

Where PyMongo does all its normal bookkeeping (error checking (ok: 0), $clusterTime parsing, cursor id parsing) and then yields the raw command response so that pymongoarrow can parse the actual documents.